packages/allure-js-commons/src/sdk/reporter/types.ts

```diff
@@ -47,10 +47,12 @@ export type LinkConfig<TOpts extends LinkTypeOptions = LinkTypeOptions> = Partia
 
 export type WriterDescriptor = [cls: string, ...args: readonly unknown[]] | string;
 
+export type GlobalLabelsConfig = Record<string, string | string[]>;
```

Comment:
BTW, if we define this thing as `Partial<Record<LabelName, string | string[]>> & Record<string, string | string[]>`, the IDE will suggest well-known label names (epic, host, etc.) to a user while still allowing them to provide custom ones.

Reply:
Good point. Pushed

Reply:
No, TypeScript erases the LabelName part, making the key type just `string`, which won't help with IDE suggestions.
You have to use type intersection to force TypeScript, keeping the LabelName part.

Reply:
You may check it yourself:

```ts
type GlobalLabelConfig = Record<LabelName | string, string | string[]>;

const v: GlobalLabelConfig = {
  /* no suggestions here */
};
```

```ts
type GlobalLabelConfig = Partial<Record<LabelName, string | string[]>> & Record<string, string | string[]>;

const v: GlobalLabelConfig = {
  /* IDE suggests well-known label names */
};
```

Reply:
It does work pretty well in idea:

<img width="1559" alt="image" src="https://github.com/user-attachments/assets/cdaf9f16-cc5b-4ff1-9695-2eded8aac734" />

But you're right, it doesn't work in LSP. So I enhanced the type

Reply:
I guess that it won't work in IDEs that use TypeScript itself to infer the types for suggestions, but it may work in IDEs that do some extra job to extract more info about the type.